modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Event.scala

```diff
@@ -28,13 +28,13 @@ object Event {
     * Events generated internally by the Engine.
     */
   sealed trait SystemEvent
-  case class Completed(i: Int) extends SystemEvent
-  case class Failed(i: Int) extends SystemEvent
+  case class Completed[R](i: Int, r: R) extends SystemEvent
+  case class Failed[E](i: Int, e: E) extends SystemEvent
   case object Executed extends SystemEvent
```

Comment:
Do `R` and `E` need to be polymorphic? I'm not saying they don't have to be, but if there is a manageable number of types of `Error`s and `Result`s it may be easier to work with concrete types.